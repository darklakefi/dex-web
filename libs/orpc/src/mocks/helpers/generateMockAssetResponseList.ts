import {
  randProductDescription,
  randProductName,
  randUuid,
  seed,
} from "@ngneat/falso";
import { type DAS, Interface, OwnershipModel } from "helius-sdk";
import { times } from "remeda";
import { FIXED_SEED } from "./constants";

export function generateAssetResponseList({
  count = 10,
}: {
  count?: number;
} = {}): DAS.GetAssetResponseList {
  seed(FIXED_SEED);
  const owner = randUuid();
  return {
    cursor: "123",
    items: times(count, (_i) => ({
      burnt: false,
      content: {
        $schema: "https://schema.dexscreener.com/asset.schema.json",
        json_uri: "https://schema.dexscreener.com/asset.schema.json",
        metadata: {
          description: randProductDescription(),
          name: randProductName(),
          symbol: randProductName().slice(0, 3).toUpperCase(),
        },
      },
      id: randUuid(),
      interface: Interface.V1NFT,
      mutable: true,
      ownership: {
        delegate: undefined,
        delegated: false,
        frozen: false,
        owner,
        ownership_model: OwnershipModel.SINGLE,
      },
    })),
    limit: 10,
    total: 10,
  };
}
