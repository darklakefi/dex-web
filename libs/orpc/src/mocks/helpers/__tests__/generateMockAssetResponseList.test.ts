import { generateAssetResponseList } from "../generateMockAssetResponseList";

describe("generateAssetResponseList", () => {
  it("should generate a list of assets", () => {
    const response = generateAssetResponseList();

    expect(response).toMatchSnapshot();
    expect(response.items.length).toBe(10);
  });
});
