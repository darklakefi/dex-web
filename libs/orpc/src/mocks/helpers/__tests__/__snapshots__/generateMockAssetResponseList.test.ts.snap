// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateAssetResponseList > should generate a list of assets 1`] = `
{
  "cursor": "123",
  "items": [
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
          "name": "Practical Wooden Ball",
          "symbol": "UNB",
        },
      },
      "id": "9e834d8e-58e5-4c10-adad-7426e48117f4",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "The Apollotech B340 is an affordable wireless mouse with reliable connectivity, 12 months battery life and modern design",
          "name": "Licensed Concrete Fish",
          "symbol": "AWE",
        },
      },
      "id": "dcaa170f-2189-4745-a6ce-d47a41ef27d0",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "The Football Is Good For Training And Recreational Purposes",
          "name": "Refined Frozen Ball",
          "symbol": "TAS",
        },
      },
      "id": "fd02d516-d3ad-4894-8cb6-932db5408464",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
          "name": "Refined Fresh Shirt",
          "symbol": "RUS",
        },
      },
      "id": "964d4e74-c94a-48db-8d72-70566658cb56",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "Ergonomic executive chair upholstered in bonded black leather and PVC padded seat and back for all-day comfort and support",
          "name": "Handcrafted Wooden Gloves",
          "symbol": "GEN",
        },
      },
      "id": "c4fd8f4b-d821-42a6-85e7-15f3ca0f1a2f",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "The slim & simple Maple Gaming Keyboard from Dev Byte comes with a sleek body and 7- Color RGB LED Back-lighting for smart functionality",
          "name": "Rustic Concrete Chips",
          "symbol": "LIC",
        },
      },
      "id": "a62dbbda-abef-4b73-a50e-6910dee76cbd",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "The beautiful range of Apple Natural√© that has an exciting mix of natural ingredients. With the Goodness of 100% Natural Ingredients",
          "name": "Generic Rubber Sausages",
          "symbol": "HAN",
        },
      },
      "id": "2919c4bb-e5b1-4909-aec3-e5963c7fdfc9",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "Carbonite web goalkeeper gloves are ergonomically designed to give easy fit",
          "name": "Sleek Wooden Bacon",
          "symbol": "AWE",
        },
      },
      "id": "9692b271-f33a-4409-a78b-4432821048e4",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "Bostons most advanced compression wear technology increases muscle oxygenation, stabilizes active muscles",
          "name": "Rustic Soft Ball",
          "symbol": "RUS",
        },
      },
      "id": "646815a5-dd44-4f9f-b722-ac277f139b0b",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
    {
      "burnt": false,
      "content": {
        "$schema": "https://schema.dexscreener.com/asset.schema.json",
        "json_uri": "https://schema.dexscreener.com/asset.schema.json",
        "metadata": {
          "description": "New range of formal shirts are designed keeping you in mind. With fits and styling that will make you stand apart",
          "name": "Handmade Frozen Salad",
          "symbol": "INC",
        },
      },
      "id": "86974372-c6c7-4e65-8c79-3c23f92d3e7e",
      "interface": "V1_NFT",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
        "ownership_model": "single",
      },
    },
  ],
  "limit": 10,
  "total": 10,
}
`;
